server/middleware/menuValidator.js

```diff
@@ -1,150 +1,104 @@
+    .not().isNumeric()
+    .withMessage('Please enter a valid menu name or menu name is short');
 };
 
+
 /**
- * @description - Validates input when updating up a menu
+ * Validate menu mealIds
  *
- * @param {object} req - HTTP Request
- * @param {object} res - HTTP Response
- * @param {function} next - Callback function
+ * @param {Object} options - Options object with key optional
  *
- * @returns {object}
+ * @returns {ValidationChain}
  */
-export const validateUpateMenu = (req, res, next) => {
-  const { menuId } = req.params;
-  const { name, mealIds } = req.body;
-  const validatedMenu = {};
-  const error = {};
-  let validMealIds;
-  let inValidMealIds;
-
-  if (menuId && (validator.isEmpty(menuId.trim()) || !validator.isNumeric(menuId))) {
-    error.menuId = 'Menu id must be a number';
-  }
-
-  if (menuId && /^[-+][0-9]*\.?/.test(menuId.trim())) {
-    error.menuId = 'Menu id cannot be less than zero';
-  }
-
-  if (menuId && /^[0-9]*\.[0-9]+$/.test(menuId.trim())) {
-    error.menuId = 'Menu id must be whole numbers';
-  }
-
-  if (menuId && (menuId > Number.MAX_SAFE_INTEGER)) {
-    error.menuId = 'Menu id is not a valid integer';
-  }
-
-  if (name && validator.isEmpty(name.trim())) {
-    error.name = 'Meal name is required';
-  }
-
-  if (name && /\d/.test(name.trim())) {
-    error.name = 'Please enter a valid meal name';
-  }
-
-  if (name && !(/(?:[a-zA-Z]+(?: [a-zA-Z]+)*){3,}/.test(name.trim()))) {
-    error.name = 'Please enter a valid meal name';
-  }
-
-  if (name) {
-    validatedMenu.name = name;
-  }
-
-  if (!mealIds) {
-    error.mealIds = 'Menu must have at least one meal';
-  }
-
-  if (mealIds && (!Array.isArray(mealIds) || (mealIds.length <= 0))) {
-    error.mealIds = 'Menu must have at least one meal';
-  }
-
-  if (mealIds && Array.isArray(mealIds)) {
-    inValidMealIds = mealIds.filter(mealId => (
-      !(validator.isNumeric(String(mealId))) || !(mealId < Number.MAX_SAFE_INTEGER)
-    ));
-    if (inValidMealIds.length > 0) {
-      error.mealIds = 'mealIds can only be integer values';
-    }
-  }
+const mealIdsValidator = ({ optional = false } = {}) => {
+  if (optional) {
+    return body('mealIds')
+      .optional()
+      .isArray().withMessage('Menu must have at least one meal')
+      .custom((mealIds) => {
+        if (mealIds.length <= 0) {
+          throw new Error('Menu must have at least one meal');
+        }
+        return true;
+      })
+      .custom((mealIds) => {
+        const invalidMealIds =
+          mealIds.filter(mealId => (
+            !Number.isInteger(mealId)
+            || mealId > Number.MAX_SAFE_INTEGER));
+
+        if (invalidMealIds.length > 0) {
+          throw new Error('mealIds can only be integer values');
+        }
+        return true;
+      });
+  }
+  return body('mealIds')
+    .exists().withMessage('Menu must have at least one meal')
+    .isArray().withMessage('Menu must have at least one meal')
```

Comment:
Good work on the validator middleware but, you can make the message 'Menu must have at least one meal' more specific, i.e in context with what the method is validating. It is validating the datatype, not the data length.

Reply:
Thank you for the feedback. Going forward, I will add more descriptive error messages that best describes the error.